titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/batch/BatchDifferenceResolver.java

```diff
@@ -238,7 +244,7 @@ public BatchDifferenceResolver(
         }
 
         Map<String, String> taskContext = getTaskContext(previousTask, unassignedIpAllocations);
-        if (kubeSchedulerPredicate.test(refJobView.getJob().getJobDescriptor())) {
+        if (KubeUtil.findFarzoneId(kubeConfiguration, refJobView.getJob()).isPresent() || kubeSchedulerPredicate.test(refJobView.getJob().getJobDescriptor())) {
```

Comment:
why not centralize this check into the `kubeSchedulerPredicate`?

Reply:
I have tried that, but the code is in a different subproject.